app/src/main/java/ch/epfl/cs311/wanderwave/navigation/NavigationActions.kt

```diff
@@ -29,6 +30,18 @@ class NavigationActions(navController: NavHostController) {
   private var _currentRouteFlow = MutableStateFlow(getCurrentRoute())
   val currentRouteFlow: StateFlow<Route?> = _currentRouteFlow
 
+  // Handle user manually clicking the back button
+  init {
+    navController.addOnDestinationChangedListener { _, destination, _ ->
```

Comment:
This part is fine, but I think you can use .let to have all in a line:
`      _currentRouteFlow.value = destination.route?.let { Route.forRouteString(it) }
`